Creating benchmarking for activations.

z-a-f opened the pull request:
This is a squash of the following PRs:

- #29427
- #29259
- #29320
- #29183


Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29182 Creating benchmarking for activations.**

Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Creating a base benchmarking class for activations.

Commit:
Update on "Creating a base benchmarking class for activations."


Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Update on "Creating a base benchmarking class for activations."


Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Update on "Creating a base benchmarking class for activations."


Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Update on "Creating a base benchmarking class for activations."


Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

mingzhe09088 commented on benchmarks/operator_benchmark/pt/qactivation_test.py:
```diff
@@ -13,13 +13,14 @@
 
-qrelu_configs = op_bench.cross_product_configs(
+qactivation_configs = op_bench.cross_product_configs(
     dims=(
```
Do you need to do cross product of all the Ranks? In the current way, only one dims will be tested.

z-a-f replied:
Oh I wanted tot check the combinations of every dimension with every other parameter. The `cross_product_configs` is not the right way to go?

z-a-f replied:
I mean, I need to check one dim at a time. For example one example to benchmarrk would be

```
dims = (1, 3, 4, 5)
permute=False
inplace=True
```

and another

```
dims = (4, 1, 1, 1)
permute=False
inplace=True
```

mingzhe09088 replied:
In this example, the cross-product function will just generate {dims} * {permute_dims} * {inplace} which is 1 * 2 * 2 = 4 different combinations. This doesn't seem to be what you want.  

If you want to generate cross product of dims, the dims needs to be split. For example, 
op_bench.cross_product_configs(
rank1 = (1,), (1, 1), (1, 1, 1),
rank2 = (2, 1), (1, 2), 
rank3 = (3, 4, 5)
...
}

This will generate cross product of {rank1} * {rank2} * {rank3} = 3 * 2 * 1 which is total of 6 different input shapes.

mingzhe09088 replied:
You are right. I misunderstood what you were trying to do here.

Commit:
Update on "Creating a base benchmarking class for activations."


Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Update on "Creating a base benchmarking class for activations."


Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Update on "Creating a base benchmarking class for activations."


Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

raghuramank100 commented on benchmarks/operator_benchmark/pt/qactivation_test.py:
```diff
@@ -6,51 +6,58 @@
+        (512,  28,  28),  # ReLU-18  # noqa
+        (512,  14,  14),  # ReLU-25  # noqa
+        # Batch = 16
+        (16,  64, 224, 224),  # ReLU-1   # noqa
```
For which operator are these configurations used?
If it is for relu/relu6 this is an overkill. I would have just 2 configs: One in 3D and one in 4D.

z-a-f replied:
One 3d and one 4d are already part of the short test below. These are longer tests for the ops -- the thing is that we noticed that the implementation is different depending on the size of the input, which is why I added all ReLU layers from the VGG. This is in total 40 tests, which translates to ~6.5 minute run (runs once every 8 hours).

I think we should keep different dimensions for benchmarking -- alternatively, we should always test with the same size.

Commit:
Update on "Creating benchmarking for activations."

This is a squash of the following PRs:

- #29427
- #29259
- #29320
- #29183



Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Update on "Creating benchmarking for activations."

This is a squash of the following PRs:

- #29427
- #29259
- #29320
- #29183



Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)

Commit:
Update on "Creating benchmarking for activations."

This is a squash of the following PRs:

- #29427
- #29259
- #29320
- #29183



Differential Revision: [D18319456](https://our.internmc.facebook.com/intern/diff/D18319456)
